ci-operator/step-registry/openshift/oc/certrotation/openshift-oc-certrotation-commands.sh

```diff
@@ -1,10 +1,56 @@
 oc adm ocp-certificates regenerate-leaf -n openshift-config-managed secrets kube-controller-manager-client-cert-key kube-scheduler-client-cert-key
 oc adm ocp-certificates regenerate-leaf -n openshift-kube-apiserver-operator secrets node-system-admin-client
 oc adm ocp-certificates regenerate-leaf -n openshift-kube-apiserver secrets check-endpoints-client-cert-key control-plane-node-admin-client-cert-key  external-loadbalancer-serving-certkey internal-loadbalancer-serving-certkey kubelet-client localhost-recovery-serving-certkey localhost-serving-cert-certkey service-network-serving-certkey
 oc adm wait-for-stable-cluster
+
+# generate new roots of trust
+oc adm ocp-certificates regenerate-top-level -n openshift-kube-apiserver-operator secrets kube-apiserver-to-kubelet-signer kube-control-plane-signer  loadbalancer-serving-signer  localhost-serving-signer service-network-serving-signer
+oc -n openshift-kube-controller-manager-operator delete secrets/next-service-account-private-key
+oc -n openshift-kube-apiserver-operator delete secrets/next-bound-service-account-signing-key
+oc adm wait-for-stable-cluster
+
+# skip the AWS STS bound SA changes because we don't have the aws CLI or know where this info is stored.
+
+# generate new client certs for kcm and ks
+oc adm ocp-certificates regenerate-leaf -n openshift-config-managed secrets kube-controller-manager-client-cert-key kube-scheduler-client-cert-key
+
+# distribute trust across all known clients
+# update our local CA bundle so that when new serving certs are used for kube-apiserver we will trust them
+oc config refresh-ca-bundle
+# produce a new kubelet bootstrap kubeconfig (used to create the first CSR and establishes ca bundle)
+oc config new-kubelet-bootstrap-kubeconfig > /tmp/bootstrap.kubeconfig
+oc whoami --kubeconfig=/tmp/bootstrap.kubeconfig --server="$(oc get infrastructure/cluster -ojsonpath='{ .status.apiServerURL }')"
+# distribute to nodes
+oc adm copy-to-node nodes --all --copy=/tmp/bootstrap.kubeconfig=/etc/kubernetes/kubeconfig
+# make all nodes use the new ca bundle
+oc adm restart-kubelet nodes --all --directive=RemoveKubeletKubeconfig
+# delete all pods to ensure that new kube-apiserver ca bundles are used by all pods
+oc adm reboot-machine-config-pool mcp/worker mcp/master
+oc adm wait-for-node-reboot nodes --all
+
+# now that trust is distributed, create new certificates using new roots
+oc adm ocp-certificates regenerate-leaf -n openshift-kube-apiserver-operator secrets node-system-admin-client
+oc adm ocp-certificates regenerate-leaf -n openshift-kube-apiserver secrets check-endpoints-client-cert-key control-plane-node-admin-client-cert-key  external-loadbalancer-serving-certkey internal-loadbalancer-serving-certkey kubelet-client localhost-recovery-serving-certkey localhost-serving-cert-certkey service-network-serving-certkey
+oc adm wait-for-stable-cluster
+
+# create new admin.kubeconfig
+oc config new-admin-kubeconfig > /tmp/shared/admin.kubeconfig
```

Comment:
@cgwalters didn't appear to work.

Reply:
We copied the PR code in our CI job step without setting `shared` folder, it ran passed, see pr https://github.com/openshift/release/pull/40684/files#diff-96847061d6dc4135ed8ec2439e36e15b28bf83c11d9fdf5e143088a67e3bf7c5R78 and job logs:
https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/logs/periodic-ci-openshift-openshift-tests-private-release-4.11-amd64-nightly-4.11-upgrade-from-stable-4.11-vsphere-ipi-proxy-fips-cert-rotation-p2-f1/1674396986943475712/artifacts/vsphere-ipi-proxy-fips-cert-rotation-p2-f1/openshift-upgrade-qe-test-cert-rotation/build-log.txt

Reply:
Hmmm.   I think this gets into a much larger confusion I have between the separation between "QE Prow jobs" and "non-QE Prow jobs" for lack of a better term...

Can I run these jobs on a PR?

But the reason things are failing here is there's something going on related to how the kubeconfig is being passed along in Prow that I don't think is being done in that QE flow.

Reply:
@cgwalters From QE side, we created and maintained a batch of test jobs for upgrade,  see https://github.com/openshift/release/tree/master/ci-operator/config/openshift/openshift-tests-private#trigger-a-new-test-for-an-existing-periodic-job-in-prow, I am not sure if  they could be executed on this PR, maybe the ENV is a little different. Will give it a try. 

The reason we copied this PR code into the PR 40684, in order to accelerate the testing of jobs in different upgrade paths and platforms.

Reply:
@deads2k Try this,
```
oc config new-admin-kubeconfig > "${SHARED_DIR}/admin.kubeconfig"
oc --kubeconfig="${SHARED_DIR}/admin.kubeconfig" whoami
```